test/e2e/storage/testsuites/subpath.go

```diff
@@ -397,7 +399,9 @@ func TestBasicSubpathFile(f *framework.Framework, contents string, pod *v1.Pod,
 // SubpathTestPod returns a pod spec for subpath tests
 func SubpathTestPod(f *framework.Framework, subpath, volumeType string, source *v1.VolumeSource, privilegedSecurityContext bool) *v1.Pod {
 	var (
-		suffix          = strings.ToLower(fmt.Sprintf("%s-%s", volumeType, rand.String(4)))
+		re = regexp.MustCompile("[^[:alnum:]]")
+		// suffix is used in pod name and it needs to be DNS-1123 label, so all strings other than alnum should be replaced with "-"
```

Comment:
Thank you for your comment.

To avoid below error to happen is the reason. I will try to explain it in the comment.
Also, I reread the below message and found out that we might need to also convert it to lower case. Because ```[[:alnum:]] == [0-9A-Za-z]```, while it requires is ```[a-z0-9]```.

```
"Pod \"pod-subpath-test-com.google.csi.gcepd-dynamicpv-hhgq\" is invalid: [spec.containers[0].name: Invalid value: \"test-container-subpath-com.google.csi.gcepd-dynamicpv-hhgq\": a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?'), spec.containers[1].name: Invalid value: \"test-container-volume-com.google.csi.gcepd-dynamicpv-hhgq\": a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?'), spec.initContainers[0].name: Invalid value: \"init-volume-com.google.csi.gcepd-dynamicpv-hhgq\": a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')]"
```

Reply:
Oh, ```ToLower()``` has already been done there.